### PR TITLE
Validation: migrate packages for esConsumes

### PR DESCRIPTION
**Pull request by sroychow**
#### PR description:
Part of #31061. Migrates validation packages to use esConsumes.

#### PR validation:
Code compiles.
#### if this PR is a backport please specify the original PR and why you need to backport that PR:
NA

### CODE REVIEW

File: Validation/RecoB/plugins/BDHadronTrackMonitoringAnalyzer.cc [hunk 1]
```diff
@@ -38,6 +38,7 @@ BDHadronTrackMonitoringAnalyzer::BDHadronTrackMonitoringAnalyzer(const edm::Para
   TrackCollectionTag_ = consumes<reco::TrackCollection>(TrackSrc_);
   PrimaryVertexColl_ = consumes<reco::VertexCollection>(PVSrc_);
   clusterTPMapToken_ = consumes<ClusterTPAssociation>(ClusterTPMapSrc_);
+  ttrackToken_ = esConsumes();
```
Comment (mmusich) [inline Validation/RecoB/plugins/BDHadronTrackMonitoringAnalyzer.cc:41]:
```suggestion
  ttrackToken_ = esConsumes(edm::ESInputTag("","TransientTrackBuilder"));
```

a label used to be specified at line 217, no?

Comment (sroychow) [reply Validation/RecoB/plugins/BDHadronTrackMonitoringAnalyzer.cc]: Yes. Thanks! I have fixed it.

**Comment by sroychow** [comment]
@cmsbuild please test

**Comment by jfernan2** [comment]
Thank you very much @sroychow !

**Comment by jfernan2** [comment]
+1

File: Validation/GlobalDigis/interface/GlobalDigisProducer.h [hunk 2]
```diff
@@ -99,7 +98,8 @@
 #include "TString.h"
 
 class PGlobalDigi;
-
+class TrackerTopology;
+class TrackerTopologyRcd;
 class GlobalDigisProducer : public edm::EDProducer {
```
Comment (Dr15Jones) [inline Validation/GlobalDigis/interface/GlobalDigisProducer.h:103]: Please update this module to be a thread-safe version.

File: Validation/GlobalHits/interface/GlobalHitsProducer.h [hunk 3]
```diff
@@ -71,7 +71,7 @@
 #include "TString.h"
 
 class PGlobalSimHit;
-
+class CaloGeometryRecord;
 class GlobalHitsProducer : public edm::EDProducer {
```
Comment (Dr15Jones) [inline Validation/GlobalHits/interface/GlobalHitsProducer.h:75]: Please update this module to a thread-safe version.

File: Validation/GlobalRecHits/interface/GlobalRecHitsProducer.h [hunk 4]
```diff
@@ -144,7 +144,8 @@
 #include "TString.h"
 
 class PGlobalRecHit;
-
+class CaloGeometryRecord;
+class TrackerTopology;
 class GlobalRecHitsProducer : public edm::EDProducer {
```
Comment (Dr15Jones) [inline Validation/GlobalRecHits/interface/GlobalRecHitsProducer.h:149]: Please update this to thread-safe version.

File: Validation/MuonRPCGeometry/plugins/RPCGeometryServTest.h [hunk 5]
```diff
@@ -2,30 +2,17 @@
...
 #include <string>
-#include <cmath>
-#include <vector>
 #include <map>
-#include <iomanip>
-#include <set>
 
+class RPCGeometry;
+class MuonGeometryRecord;
 class RPCGeometryServTest : public edm::EDAnalyzer {
```
Comment (Dr15Jones) [inline Validation/MuonRPCGeometry/plugins/RPCGeometryServTest.h:16]: Please update to thread-safe version.

**Comment by sroychow** [comment]
@Dr15Jones I made the modules edm::one type. Do you think it is okay?

**Comment by jfernan2** [comment]
please test

**Comment by cvuosalo** [comment]
The comparison differences look like the problem recently fixed by #34409. Let's try the tests again.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**